File: src/main/kotlin/com/huawei/cangjie/psi/CjTypeConstraintList.java
package com.huawei.cangjie.psi;

import com.huawei.cangjie.psi.stubs.CangJiePlaceHolderStub;
import com.intellij.lang.ASTNode;
import org.jetbrains.annotations.NotNull;

import java.util.List;

import  com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes;
public class CjTypeConstraintList extends CjElementImplStub<CangJiePlaceHolderStub<CjTypeConstraintList>> {
    public CjTypeConstraintList(@NotNull ASTNode node) {
        super(node);
    }

    public CjTypeConstraintList(@NotNull CangJiePlaceHolderStub<CjTypeConstraintList> stub) {
        super(stub, CjStubElementTypes.TYPE_CONSTRAINT_LIST);
    }

    @Override
    public <R, D> R accept(@NotNull CjVisitor<R, D> visitor, D data) {
        return visitor.visitTypeConstraintList(this, data);
    }



    @NotNull
    public List<CjTypeConstraint> getConstraints() {
        return getStubOrPsiChildrenAsList(CjStubElementTypes.TYPE_CONSTRAINT);
    }
}

File: src/main/kotlin/com/huawei/cangjie/psi/CjTypeConstraintList.kt
package com.huawei.cangjie.psi

import com.huawei.cangjie.psi.stubs.CangJiePlaceHolderStub
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes
import com.intellij.lang.ASTNode

class CjTypeConstraintList : CjElementImplStub<CangJiePlaceHolderStub<CjTypeConstraintList > > {
    constructor(node: ASTNode) : super(node)

    constructor(stub: CangJiePlaceHolderStub<CjTypeConstraintList >) : super(
        stub,
        CjStubElementTypes.TYPE_CONSTRAINT_LIST
    )

    override fun <R, D> accept(visitor: CjVisitor<R, D>, data: D?): R {
        return visitor.visitTypeConstraintList(this, data)
    }


    val constraints: List<CjTypeConstraint>
        get() = getStubOrPsiChildrenAsList(
            CjStubElementTypes.TYPE_CONSTRAINT
        )
}



